pkg/recipes/terraform/config/providers/aws.go

```diff
@@ -132,9 +143,14 @@ func (p *awsProvider) generateProviderConfigMap(credentials *credentials.AWSCred
 		config[awsRegionParam] = region
 	}
 
-	if credentials != nil && credentials.AccessKeyID != "" && credentials.SecretAccessKey != "" {
-		config[awsAccessKeyParam] = credentials.AccessKeyID
-		config[awsSecretKeyParam] = credentials.SecretAccessKey
+	if credentials != nil && credentials.Kind == ucp_datamodel.AWSAccessKeyCredentialKind && credentials.AccessKeyCredential != nil &&
+		credentials.AccessKeyCredential.AccessKeyID != "" && credentials.AccessKeyCredential.SecretAccessKey != "" {
+		config[awsAccessKeyParam] = credentials.AccessKeyCredential.AccessKeyID
+		config[awsSecretKeyParam] = credentials.AccessKeyCredential.SecretAccessKey
+	}
+
+	if credentials != nil && credentials.Kind == ucp_datamodel.AWSIRSACredentialKind && credentials.IRSACredential != nil && credentials.IRSACredential.RoleARN != "" {
+		config[awsRoleARN] = credentials.IRSACredential.RoleARN
 	}
```

Comment:
We can move `if credentials != nil` outside the two if loops, something like this:

```suggestion
if credentials != nil {
	if credentials.Kind == ucp_datamodel.AWSAccessKeyCredentialKind {
	    if credentials.AccessKeyCredential != nil &&
		credentials.AccessKeyCredential.AccessKeyID != "" && credentials.AccessKeyCredential.SecretAccessKey != "" {
		config[awsAccessKeyParam] = credentials.AccessKeyCredential.AccessKeyID
		config[awsSecretKeyParam] = credentials.AccessKeyCredential.SecretAccessKey
             }
	} else if credentials.Kind == ucp_datamodel.AWSIRSACredentialKind {
	        if credentials.IRSACredential != nil && credentials.IRSACredential.RoleARN != "" {
		config[awsRoleARN] = credentials.IRSACredential.RoleARN
           }
        } else {
         // Error out for unexpected credentials kind? Is this possible at this point?
        }
   }
```

Reply:
I added the change. The else part is not neccessary, since if we PUT a credential with any kind other than these two then we will get an error 
```
$.properties.kind in body should be one of [AccessKey IRSA]"
```

Reply:
makes sense, thanks